### PR TITLE
LV624 Fog Removal + River Change

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

Several major but interconnected changes have been made to 624.

All fog on the map is removed, meaning the entire map is accessible from the start.

The river flowing through the centre of the map inflicts significant slowdown to any mob that tries to move through it, making it extremely dangerous to traverse it while enemies are in range.

Three new bridges have been placed around the map (one by LZ2 comms, one north of hydro, one by eastern river), the bridges are guaranteed to be present and are fully intact, offering a traversable area to move through.

The areas around these bridges have been expanded for xeno weeding.

Nightmare inserts regarding fog or river bridges have been removed from the map.

I've removed the floodlights in tfort and replaced them with the breakable flood lights, so xenos can kill the lights instead of them being on forever

# Explain why it's good for the game

The Fog is a ancient 2016 concept that we have long passed needing.

My changes are aimed at shaking up the meta of a map that has stagnated for a decade. The addition of three bridges, plus the massive slowdown of the river, will crate artificial chock points that the Marines and Xenos will want to fight over to establish control over the enemies territory. While the river can still be crossed, it represents a death sentence if caught by a enemy. 

Fruthermore the extreme slowdown will encourage both sides to take and hold a bridge.

The expanded weeding grounds of the bridges will allow xenos to establish better river defences and even allow them to confront the marines in the south river area on first contact. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Screenshot 2023-10-22 18 00 59

Screenshot 2023-10-22 18 01 07

Screenshot 2023-10-22 18 01 26

</details>

# Changelog
:cl:
maptweak: Fog on 624 has been removed
maptweak: River on 624 now inflicts a massive movement slowdown on anyone cross it
maptweak: Three bridges have been placed around 624 allowing easy access across the river
maptweak: Expanded the areas xenos can weed around the river to allow better defences. 
maptweak: Flood lights around t-fort are now breakable by xenos
/:cl:

### CODE REVIEW

**Comment by Wintermote** [comment]
A useful option might be a buildable 'bridge' turf tile; it'd be useful not just for this LV but also trijent and any other map with crossable water or dangerous terrains

**Comment by fira** [comment]
I see you only changed the main map file. Did you test this with nightmare? It includes fog modifications.

**Comment by Steelpoint** [comment]
I removed the nightmare insert spawns that affect an area that has fog

**Comment by fira** [comment]
You need to remove the config for it too then, this will cause errors on startup since it'll still try to load them without finding where

**Comment by cuberound** [comment]
adding more survivor to north part of river would be also cool

**Comment by fira** [comment]
Reopened once due to recent difficulties in map reviewing

**Comment by Nanu308** [comment]
Requested @Zonespace27 input, if he wants the gameplay implications for it, i can go over the mapping itself

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by Lagomorphica** [comment]
While I understand a desire to shake up and change how maps flow I think this kind of misses the mark and if anything rebalances the outside of the caves to be even more in marines favour - e.g, encourages sitting in the caves more and reduces the amount of beach fights and holds in general, or skirmishing on the river/in the colony due to adding even less reasons for xenomorphs to do it.

# What I think the issue is

As it stands, slows are frustrating for both sides. This is exacerbated even more by the fact that an entire midsection of the map is in a constant state of forced, perpetual slowdown that cannot be counteracted aside from using bridges - which force xenomorphs to use them more than marines. Marines in the river are granted safety and practical immunity to all castes aside from spitters and praetorians due to their ability to use ranged attacks, as attempting to push across to attack them exposes them to fire from the ranged group that can also move while firing. Engaging with a dash also poses the same issue of having to retreat across the water once you are done and denies you an escape tool once you are done.

Skirmishing on the bridges is actively disadvantageous for them as well since marines can simply destroy fortifications from range with CAS and mortar and walk along the sides of it, and through the river - which endangers any castes that rely on walking up to marines to engage them and offers cover to the marines. Backliners attempting to flee are forced to use predictable, often fortified areas, or else they suffer a slowdown that enables most marines to catch up to them (unless they have a pounce on cooldown). This is also exacerbated by both communication towers being on the other side of the river, and while they have a bridge adjacent to them, it is vastly easier for marines to hold these chokes than it is for xenomorphs to push them - and it is also vastly easier for marines to flank around the water. Tunnels are an option, yes, but the areas that can be tunnelled too also offer little in the way of protection (weeds or structures being unbuildable on large swathes of grass, for instance).

This could possibly be fixed by adding the random flings that are on New Varadero water, but it would also cause more problems in terms of half of the map (outside of three artificial choke points) heavily altering your movement and making it even less fun and enjoyable to move through for both sides with no real difficulty. You might get great RNG and pass through it completely unharmed or immediately be flung into a warrior/marine and killed with zero counterplay available. 

The core issue with all of this, though, is that it destroys part of the playable area. It is no longer truly viable to skirmish near the beach for xenomorphs, and being pushed back to it is almost guaranteed to be a death sentence - whereas marine routs are safer because they are granted far more safety. Most of LV-624 is already unused outside of endgame situations and this pushes it further, alongside making it harder to force those endgame situations, or to end a stalemate nuke as xeno. 

The additional weedable area near hydroponics also does not justify the greater difficulty to moving around the map as a xenomorph, as it stands, since once the bridge is lost it is almost never going to be retaken short of a marine push immediately faltering on the beach.

The light posts on either side of the bridges also clip through the ledges and, while not a balance issue, it is visually distracting. While also more of a personal critique the addition of a flood gate moves from the river being a mostly natural feature to being implied as more of a controlled element compared to the oppressing nature of the jungle around it, but that's rather irrelevant to gameplay as well.

# Overall

As much as I may have repeated myself, my core issue here is that it encourages marines to sit in the water and encourages xenos to actively avoid interacting with the colony at all. It favours marines far more than it changes up the gameplay loop and makes backlining, frontlining, and endgame assaults harder by changing how the map flows and granting more territory control to marines by making only four viable lanes of attack for xenomorphs if marines are near the river at all, while allowing marines to push across all of it and to attack from it in safety. 

Adding more features to make the river more dangerous may alleviate this although it would still keep the issue of the marines being a side that has multiple ranged attacks and numerous tools for controlling and breaking apart non-natural chokepoints into flat and open ground that is difficult to assault, while also limiting possibilities for how xenos can address this (flanking, taking different routes) by making tunnels one of the few viable ways to assault communications. Committing to one side of the colony is also assuredly going to be a death sentence as it is impossible to quickly and easily leave once you have crossed - unless you are a marine, in which case you are granted safety from being pursued. 

While it is more dangerous for a marine caught alone as well in the river and they say, get warrior lunged, even without the slowdown they would be in the same danger as prior to this PR.

TL;DR -
- More dangerous for xenomorphs than marines to push the river.
- Provides natural soft-cover for marines by disincentivizing xenos to push.
- Draws out and stagnates rounds by reducing playable area.
- Communication towers are easier to hold by marines due to this and it is easier to nuke.
- Committing and attempting to win the game as a xenomorph is harder as if you have to retreat your options are going to be heavily telegraphed while marines do not suffer the same issue as they can attack from range and while inside of the river, forcing you to move towards them and be slowed.

Personally I am not sure on how to fix the issue while keeping the spirit of the PR intact, although that is likely better handled by somebody with a better idea of gameplay flow than myself.

**Comment by Steelpoint** [comment]
I hate merge conflicts, I might revisit the idea in the future.

**Comment by ihatethisengine** [comment]
tbh revert slowdown, the rest of the changes are kinda ok

**Comment by Steelpoint** [comment]
I've slept on it and I think I'll redo the PR in a seperate one and have everything except the slowdown as @ihatethisengine said
